src/Lecture3.hs

```diff
@@ -125,9 +137,14 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+    (<>) :: Reward -> Reward -> Reward
+    r <> s =
+        Reward (rewardGold r <> rewardGold s)
+               (rewardSpecial r || rewardSpecial s)
 
 instance Monoid Reward where
+    mempty :: Reward
+    mempty = Reward (Gold 0) False
```

Comment:
You can use `mempty` for `Gold` here because it already has a `Monoid` instance 🏅 
```suggestion
    mempty = Reward mempty False
```